crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -314,6 +314,51 @@ impl Schedule {
             system.apply_buffers(world);
         }
     }
+
+    /// Get the state of system-stepping
+    pub fn stepping(&self) -> bool {
+        self.executor.stepping()
+    }
+
+    /// Get the name of the next system that will be run in the
+    /// schedule if stepping is enabled
+    pub fn next_system(&self) -> Option<String> {
+        self.executor
+            .next_system()
+            .map(|idx| self.executable.systems[idx].name().to_string())
+    }
+
+    /// Enable or disable system-stepping mode
+    pub fn set_stepping(&mut self, value: bool) {
+        self.executor.set_stepping(value);
+    }
+
+    /// Run the next system when in stepping-mode
+    pub fn step_system(&mut self) {
+        self.executor.step_system();
+    }
+
+    /// Run any remaining systems for this frame, then stop before running any systems for the next
+    /// frame.
+    ///
+    /// If some of the systems for this frame have already been run by calling
+    /// [`step_system`](crate::Schedule::step_system), they will not be run a second time when this
+    /// method is called.
+    pub fn step_frame(&mut self) {
+        self.executor.step_frame();
+    }
```

Comment:
These methods are left over from before the addition of  `ScheduleEvent` to control stepping.  They're still used in the `Schedule` stepping tests, but aren't used elsewhere.

Probably should remove these, and use `Schedule::handle_event()` in the `Schedule` stepping tests.

Reply:
Agreed.